fuel-core/src/schema/dap.rs

```diff
@@ -192,4 +192,259 @@ impl DapMutation {
+            })
+        }
+    }
+
+    #[cfg(not(feature = "debug"))]
+    async fn continue_tx(
+        &self,
+        _ctx: &Context<'_>,
+        _id: ID,
+    ) -> async_graphql::Result<self::gql_types::RunResult> {
+        Err(async_graphql::Error::new(
+            "Feature 'debug' is not compiled in",
+        ))
+    }
+
+    #[cfg(feature = "debug")]
+    async fn continue_tx(
+        &self,
+        ctx: &Context<'_>,
+        id: ID,
+    ) -> async_graphql::Result<self::gql_types::RunResult> {
+        trace!("Continue execution of VM {:?}", id);
+
+        let mut locked = ctx.data_unchecked::<GraphStorage>().lock().await;
+        let vm = locked
+            .vm
+            .get_mut(&id)
+            .ok_or_else(|| async_graphql::Error::new("VM not found"))?;
+        let state = match vm.resume() {
+            Ok(state) => state,
+            // The transaction was already completed earlier, so it cannot be resumed
+            Err(fuel_vm::error::InterpreterError::DebugStateNotInitialized) => {
+                return Ok(self::gql_types::RunResult {
+                    state: self::gql_types::RunState::Completed,
+                    breakpoint: None,
+                })
+            }
+            // The transaction was already completed earlier, so it cannot be resumed
+            Err(err) => return Err(async_graphql::Error::new(format!("VM error: {err:?}"))),
+        };
+
+        let dbgref = state.debug_ref();
+
+        Ok(self::gql_types::RunResult {
+            state: match dbgref {
+                Some(_) => self::gql_types::RunState::Breakpoint,
+                None => self::gql_types::RunState::Completed,
+            },
+            breakpoint: dbgref.and_then(|d| match d {
+                DebugEval::Continue => None,
+                DebugEval::Breakpoint(bp) => Some(bp.into()),
+            }),
+        })
+    }
+}
+
+mod gql_types {
+    //! GraphQL type wrappers
+    use std::str::FromStr;
+
+    use async_graphql::*;
+
+    use crate::schema::scalars::U64;
+
+    #[cfg(feature = "debug")]
+    use fuel_vm::prelude::Breakpoint as FuelBreakpoint;
+
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+    pub struct ContractId(fuel_types::ContractId);
+
+    #[Scalar]
+    impl ScalarType for ContractId {
+        fn parse(value: Value) -> InputValueResult<Self> {
+            if let Value::String(value) = &value {
+                Ok(Self(fuel_types::ContractId::from_str(value)?))
+            } else {
+                Err(InputValueError::expected_type(value))
+            }
+        }
+
+        fn to_value(&self) -> Value {
+            Value::String(self.0.to_string())
+        }
+    }
+
+    impl From<fuel_types::ContractId> for ContractId {
+        fn from(id: fuel_types::ContractId) -> Self {
+            Self(id)
+        }
+    }
+    impl From<ContractId> for fuel_types::ContractId {
+        fn from(id: ContractId) -> Self {
+            id.0
+        }
+    }
```

Comment:
This should already exist in scalars

```suggestion
    pub use crate::schema::scalars::ContractId;
```

Reply:
Done